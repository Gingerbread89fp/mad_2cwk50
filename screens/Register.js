import React, { Component } from 'react';
import { View, Text, Alert, TouchableOpacity } from 'react-native';
import CustomFormInput from '../app_components/inputField'

import styles from '../styles/app_style'

class Register extends Component{
  constructor(props){
    super(props);
    this.state = {
        given_name: '',
        family_name: '',
        email: '',
        password: ''
    }
  }

  static navigationOptions={
    title: 'Registration page',
    headerTitleStyle: styles.page_title,
    headerStyle: {height: 64, marginBottom: 12},
  }

  register(){
    return fetch('http://10.0.2.2:3333/api/v0.0.5/user', {
      method: 'POST',
      headers: { "Content-Type": "application/json", 'Accept': 'application/json', },
      body: JSON.stringify({
        given_name: this.state.given_name,
        family_name: this.state.family_name,
        email: this.state.email,
        password: this.state.password
      })
    })
    .then((response) => {
      Alert.alert("Registered successfully")
    })
    .then((response) => { this.props.navigation.navigate('Login')})
    .catch((error)=>{
      console.log(error)
    })
  }

  render(){
    return(
      <View style={styles.details_form}>

        <CustomFormInput 
            labelTitle={'First Name:'}
            value={this.state.given_name}
            onChangeText={(given_name) => this.setState({given_name})}
        />

        <CustomFormInput 
            labelTitle={'Last Name:'}
            value={this.state.family_name}
            onChangeText={(family_name)=>this.setState({family_name})}
        />

        <CustomFormInput 
            labelTitle={'Email:'}
            value={this.state.email}
            onChangeText={(email)=>this.setState({email})}
        />

        <CustomFormInput 
            labelTitle={'Password:'}
            value={this.state.password}
            onChangeText={(password)=>this.setState({password})}
        />
        
        <TouchableOpacity
          style={styles.button_style} 
          onPress={()=> this.register()}>
              <Text>SIGNUP</Text>
        </TouchableOpacity>

      </View>
    )

  }



}

export default Register;