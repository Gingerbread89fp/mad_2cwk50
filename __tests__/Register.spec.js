import React from 'react';
import renderer from 'react-test-renderer';
import Register from '../screens/Register';

describe('<Register />', ()=>{

    test('render Register page correctly', ()=>{
        const reg = renderer.create(<Register />).toJSON()
        expect(reg).toMatchSnapshot();
    })

})