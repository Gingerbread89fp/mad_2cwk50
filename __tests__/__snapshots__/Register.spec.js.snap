// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> render Register page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#B9B8D3",
      "height": 560,
    }
  }
>
  <View
    style={
      Object {
        "alignContent": "space-between",
        "backgroundColor": "white",
        "borderColor": "#1F5673",
        "borderRadius": 8,
        "borderStyle": "solid",
        "borderWidth": 1,
        "elevation": 4,
        "height": 500,
        "marginHorizontal": 12,
        "marginVertical": 8,
        "paddingTop": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingTop": 18,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
           
          First Name:
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#90C3C8",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 4,
              "height": 48,
              "marginBottom": 20,
              "paddingLeft": 10,
              "width": 360,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
           
          Last Name:
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#90C3C8",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 4,
              "height": 48,
              "marginBottom": 20,
              "paddingLeft": 10,
              "width": 360,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
           
          Email:
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          style={
            Object {
              "borderColor": "#90C3C8",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 4,
              "height": 48,
              "marginBottom": 20,
              "paddingLeft": 10,
              "width": 360,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
           
          Password:
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#90C3C8",
              "borderRadius": 8,
              "borderStyle": "solid",
              "borderWidth": 4,
              "height": 48,
              "marginBottom": 20,
              "paddingLeft": 10,
              "width": 360,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#90C3C8",
            "borderColor": "#1F5673",
            "borderRadius": 8,
            "borderWidth": 2,
            "height": 40,
            "justifyContent": "center",
            "marginTop": 18,
            "opacity": 1,
            "textDecorationColor": "black",
            "width": 164,
          }
        }
      >
        <Text>
          SIGNUP
        </Text>
      </View>
    </View>
  </View>
</View>
`;
